test/test_utils/ec2.py

```diff
@@ -257,6 +269,43 @@ def launch_instance(
     return response["Instances"][0]
 
 
+@retry(
+    reraise=True,
+    stop=stop_after_delay(30 * 60),  # Keep retrying for 30 minutes
+    wait=wait_random_exponential(min=60, max=5 * 60),  # Retry after waiting 1-5 minutes
+)
+def launch_instances_with_retry(
+    ec2_resource, availability_zone_options, ec2_create_instances_definition
+):
+    """
+    Helper function to launch EC2 instances with retry capability, to allow multiple attempts
+    when facing instance capacity issues.
+    :param ec2_resource: boto3 EC2 Service Resource object
+    :param availability_zone_options: list of availability zones in which to try to run instances
+    :param ec2_create_instances_definition: dict of parameters to pass to
+        ec2_resource.create_instances
+    :return: list of EC2 Instance Resource objects for instances launched
+    """
+    instances = None
+    if availability_zone_options:
+        error = None
+        for a_zone in availability_zone_options:
+            ec2_create_instances_definition["Placement"] = {"AvailabilityZone": a_zone}
+            try:
+                instances = ec2_resource.create_instances(**ec2_create_instances_definition)
+                if instances:
+                    break
+            except ClientError as e:
+                LOGGER.error(f"Failed to launch in {a_zone} due to {e}")
+                error = e
+                continue
```

Comment:
nit: Do we need a `continue` here? Will it not proceed with the next `a_zone` by default?

Reply:
Good point. Yes, it will. I think that this is an artifact leftover from the code refactor.

As we discussed offline, let's leave this here for now so that we don't need another cycle of testing.